src/github.com/getlantern/checkfallbacks/verify.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/getlantern/flashlight/client"
+)
+
+var (
+	// https://github.com/getlantern/lantern/issues/3147
+	redirectSites = []string{
+		// sites that redirect to http when accessed through https
+		"http://www.bbc.co.uk",
+		"http://www.speedtest.net",
+		// sites that redirect to itself if the "Host:" HTTP header contains port
+		"http://lowendbox.com",
+		"http://sourceforge.net",
+		"http://krypted.com/mac-security/manage-profiles-from-the-command-line-in-os-x-10-9/",
+	}
+
+	atsDefaultBody = []byte(`<HTML>
+<HEAD>
+<TITLE>Not Found on Accelerator</TITLE>
+</HEAD>
+
+<BODY BGCOLOR="white" FGCOLOR="black">
+<H1>Not Found on Accelerator</H1>
+<HR>
+
+<FONT FACE="Helvetica,Arial"><B>
+Description: Your request on the specified host was not found.
+Check the location and try again.
+</B></FONT>
+<HR>
+</BODY>
+`)
+)
+
+func verifyFallback(fb *client.ChainedServerInfo, c *http.Client) {
+	verifyMimicWhenNoAuthToken(fb, c)
+	verifyMimicWithInvalidAuthToken(fb, c)
+	for _, s := range redirectSites {
+		verifyRedirectSites(fb, c, s)
+	}
+}
+
+func verifyMimicWhenNoAuthToken(fb *client.ChainedServerInfo, c *http.Client) {
+	req, err := http.NewRequest("GET", "http://www.google.com/humans.txt", nil)
+	if err != nil {
+		log.Errorf("%v: NewRequest() error : %v", fb.Addr, err)
+		return
+	}
+	// intentionally not set auth token
+	doVerifyMimic(fb.Addr, req, c)
+}
+
+func verifyMimicWithInvalidAuthToken(fb *client.ChainedServerInfo, c *http.Client) {
+	req, err := http.NewRequest("GET", "http://www.google.com/humans.txt", nil)
+	if err != nil {
+		log.Errorf("%v: NewRequest() error : %v", fb.Addr, err)
+		return
+	}
+	req.Header.Set("X-LANTERN-AUTH-TOKEN", "invalid")
+	doVerifyMimic(fb.Addr, req, c)
+}
+
+func doVerifyMimic(addr string, req *http.Request, c *http.Client) {
+	resp, err := c.Do(req)
+	if err != nil {
+		log.Errorf("%v: requesting humans.txt failed: %v", addr, err)
+		if *verbose {
```

Comment:
Oh I didn't notice that I'm printing the original request. In fact it's somewhat meaningless to print request here if http connection failed.